packages/waku/src/router/create-pages.ts

```diff
@@ -755,29 +775,37 @@ export const new_createPages = <
     renderRoute: async (path) => {
       await configure();
 
+      // path without slugs
+      const routePath = getRoutePath(path);
+      if (!routePath) {
+        throw new Error('Route not found: ' + path);
+      }
+
       const pageComponent = (staticComponentMap.get(
-        joinPath(path, 'page').slice(1), // feels like a hack
-      ) ?? dynamicPagePathMap.get(path)?.[1])!;
+        joinPath(routePath, 'page').slice(1), // feels like a hack
+      ) ?? dynamicPagePathMap.get(routePath)?.[1])!;
 
       const result: Record<string, ReactNode> = {
         root: createElement(
           rootItem ? rootItem.component : DefaultRoot,
           null,
           createElement(Children),
         ),
-        [`page:${path}`]: createElement(
+        [`page:${routePath}`]: createElement(
           pageComponent,
           null,
           createElement(Children),
         ),
```

Comment:
@dai-shi the page props are not being passed into the page component on the server which causes the test to fail. 

this is something that I was assuming defineRouter would be responsible for. Any thoughts?

Reply:
Page props are `slug` names, right? It's called `mapping` in the code.
No, it's new_createPages that is responsible for it.

I missed it too, but we need a wrapper component like this:
https://github.com/dai-shi/waku/blob/33bea3d603067c5bc0b64a9db78a41c3a5aff800/packages/waku/src/router/create-pages.ts#L430-L431

Reply:
I partially understand this, but some more explanation might be needed here.

`WrappedComponent` is a `FunctionComponent` and the return type of `renderRoute` is `Record<string, ReactNode>`

If we use WrappedComponent as the return for `page${routePath}`, should the return type be changed to `Record<string, ReactNode | FunctionComponent>` or did you mean to do something different entirely?

----

A separate thing I noticed is that `getPathConfig` has the mapping for each of the static paths. I did a naive attempt to get the mapping here:

https://github.com/tylersayshi/waku/blob/refactor-e2e-test-createPages/packages/waku/src/router/create-pages.ts#L788-L789

this is very clearly wrong when I test it, how would you do this here?

Reply:
You are right. `renderRoute` returns `Record<string, ReactNode>`, so we need to create the mapping in the `renderRoute` function. I'm not sure if `getPathConfig` works, but if you get the requirement, I'll leave it to you for now. Let me know if you need any further help.

Reply:
ah I just had a typo haha - that was way simpler to fix than I expected :)